keps/sig-testing/1746-stage-e2e-test-framework/README.md

```diff
@@ -0,0 +1,610 @@
+  directory.
+
+
+## Summary
+
+<!--
+This section is incredibly important for producing high quality user-focused
+documentation such as release notes or a development roadmap.  It should be
+possible to collect this information before implementation begins in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself.  KEP editors, SIG Docs, and SIG PM
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+-->
+
+In order to improve the user experience of all the people who have to maintain
+e2e tests in Kubernetes and with the aim of reducing technical debt in e2e tests
+(i.e., have more organized tests, better utility functions) we propose to work
+towards moving the e2e test framework from its current place in
+[kubernetes/test/e2e/framework](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/framework)
+to the appropriate location under `k8s.io/kubernetes/staging/src/k8s.io/e2e-framework`.
+
+Our main goal is to improve the experience of maintainers and building on top
+of what we already have instead of starting from scratch.
+
+Moving the e2e test framework into staging means that the e2e test framework
+will be used and maintained as a component independent of core Kubernetes.
+By doing this, the e2e test framework will only make use of public APIs and
+will be self-contained.
+This will help maintainers of e2e tests have tools to easily test Kubernetes
+any component that run on Kubernetes.
+This will also help external consumers of the e2e test framework by reducing
+the amount of dependency management they have to do now when importing the e2e
+test framework.
+
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this KEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+The e2e test framework is a Kubernetes component that has been evolving
+organically to meet all the needs that contributors have in order to test
+enhancements and behaviours of Kubernetes.
+The SIG Testing sub-project
+[Testing Commons](https://github.com/kubernetes/community/tree/master/sig-testing#testing-commons)
+has been looking at ways of improving the user experience for contributors to
+the Kubernetes community.
+This KEP is part of that goal.
+
+We foresee the following benefits coming out of this work:
+
+1. Maintainability: the e2e test framework lacks a general API - there has been
+  little discussion on what parts of the framework do what.
+  From this work we plan to organize the code that is present and use only
```

Comment:
is there a reason staging is necessary to organize the code?

Reply:
The proposal to move to staging is to address the point below about external consumption.